pkg/sign/certificate_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sign
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"encoding/pem"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/sigstore/sigstore-go/pkg/testing/ca"
+)
+
+var virtualSigstore *ca.VirtualSigstore
+
+func getFulcioResponse() (*http.Response, error) {
+	var err error
+	if virtualSigstore == nil {
```

Comment:
Mutating a package level var can result in bugs if we go add another test that uses this same var (as tests can run in parallel), so I would guard this check with [`sync.Once`](https://pkg.go.dev/sync#Once.Do), which is like a single-use mutex.